stage1/rootfs/prepare-app/prepare-app.c

```diff
@@ -179,5 +179,15 @@ int main(int argc, char *argv[])
 	pexit_if(symlink("/dev/pts/ptmx", to) == -1,
 		"Failed to create /dev/ptmx symlink");
 
+	/* diagexec */
+	exit_if(snprintf(to, sizeof(to), "%s/.diagexec", root) >= sizeof(to),
+		"Path too long: \"%s\"", to);
+	fd = open(to, O_WRONLY|O_CREAT|O_CLOEXEC|O_NOCTTY, 0755);
+	if (fd != -1)
+		close(fd);
+	pexit_if(mount("/diagexec", to, "bind", MS_BIND, NULL) == -1,
```

Comment:
Ideally we can just own a directory, perhaps something to add to the ACI spec for Linux. Something like /appc or something that is owned exclusively by the executor for supporting binaries.